### PR TITLE
Fix broken recovery test

### PR DESCRIPTION
**Pull request by RahulKushwaha**
**Background**: `ib_recover` is broken at two places: 
1. Checking the desired file size during recovery. 
2. Checking the permissions of the directory/file.

**Fixes**: 
1. Fixed the permission check by checking if we have owner read and owner write permissions. 
2. Correctly estimating the file size.

### CODE REVIEW

File: srv/srv0start.cc [hunk 1]
```diff
@@ -612,13 +606,17 @@ static db_err open_or_create_log_file(
     ret = os_file_get_size(files[i], &size);
     ut_a(ret);
 
-    if (size != (off_t) srv_log_file_size) {
+    off_t size_high = srv_calc_high32(srv_log_file_size);
+    off_t file_size = (off_t)srv_calc_low32(srv_log_file_size) + (((off_t)size_high) << 32);
+    if (size != file_size) {
```
Comment (RahulKushwaha) [inline srv/srv0start.cc:611]: @sunbains Does this look right? Without this we are comparing 2048 with 32 mb.

Comment (sunbains) [reply srv/srv0start.cc:611]: I think you are right, it's converted to  number of pages from number of bytes in  "db_err srv_normalize_init_values()", better to compare it with srv_log_file_curr_size then which is in bytes.

Comment (RahulKushwaha) [reply srv/srv0start.cc]: Makes sense! Fixed now :)

**Comment by sunbains** [comment]
I fixed this as part of the recovery refactoring.

**Pull request closed without merging**